File: fixtures/scale/swift/large/Families/Vigenere/VigenereBench.swift
// Benchmark entry point for the vigenere family; wraps BenchmarkEngine
// with this family's cipher and known-answer vectors.
struct VigenereBench {
    private let engine = BenchmarkEngine()

    func measure(iterations: Int) -> BenchSample {
        let cipher = VigenereCipher()
        let vectors = VigenereVectors.all()
        return engine.sample(cipher, vectors, iterations)
    }
}

File: fixtures/scale/php/large/src/Families/Vigenere/VigenereBench.php
<?php

namespace CipherLab\Families\Vigenere;

use CipherLab\Abstractions\BenchSample;
use CipherLab\Engine\BenchmarkEngine;

// Benchmark entry point for the vigenere family; wraps BenchmarkEngine
// with this family's cipher and known-answer vectors.
class VigenereBench
{
    private BenchmarkEngine $engine;

    public function __construct()
    {
        $this->engine = new BenchmarkEngine();
    }

    public function measure(int $iterations): BenchSample
    {
        return $this->engine->sample(new VigenereCipher(), VigenereVectors::all(), $iterations);
    }
}
